apps/dh/api-dh/source/DataHub.WebApi/Modules/ProcessManager/Requests/RequestOperations.cs

```diff
@@ -0,0 +1,154 @@
+{
+    [Query]
+    [UsePaging]
+    [UseSorting]
+    // TODO: This query should be authorized
+    public static Task<IEnumerable<IActorRequestQueryResult>> GetRequestsAsync(
+        IRequestsClient client) => client.GetRequestsAsync();
+
+    [Query]
+    public static async Task<RequestOptions> GetRequestOptionsAsync(
+        IHttpContextAccessor httpContextAccessor,
+        IMarketParticipantClient_V1 marketParticipantClient)
+    {
+        // TODO: Create common functionality for this use case
+        var user = httpContextAccessor.HttpContext?.User;
+        var associatedActor = user?.GetAssociatedActor()
+            ?? throw new InvalidOperationException("No associated actor found.");
+
+        var selectedActor = await marketParticipantClient.ActorGetAsync(associatedActor);
+        return new RequestOptions(user, selectedActor.MarketRole.EicFunction);
+    }
+
+    public class RequestOptions(ClaimsPrincipal user, EicFunction marketRole)
+    {
+        public IEnumerable<Option<CalculationType>> GetCalculationTypes()
+        {
+            var calculationTypes = new List<Option<CalculationType>>();
+
+            if (user.HasRole("request-aggregated-measured-data:view"))
+            {
+                calculationTypes.Add(new Option<CalculationType>(CalculationType.Aggregation));
+                calculationTypes.Add(new Option<CalculationType>(CalculationType.BalanceFixing));
+            }
+
+            if (user.HasRole("request-wholesale-settlement:view"))
+            {
+                calculationTypes.Add(new Option<CalculationType>(CalculationType.WholesaleFixing));
+                calculationTypes.Add(new Option<CalculationType>(CalculationType.FirstCorrectionSettlement));
+                calculationTypes.Add(new Option<CalculationType>(CalculationType.SecondCorrectionSettlement));
+                calculationTypes.Add(new Option<CalculationType>(CalculationType.ThirdCorrectionSettlement));
+            }
+
+            return calculationTypes;
+        }
+
+        public IEnumerable<Option<MeteringPointType>> GetMeteringPointTypes()
+        {
+            var meteringPointTypes = new List<Option<MeteringPointType>>
+            {
+                new Option<MeteringPointType>(MeteringPointType.FlexConsumption),
+                new Option<MeteringPointType>(MeteringPointType.NonProfiledConsumption),
+                new Option<MeteringPointType>(MeteringPointType.Production),
+            };
+
+            if (marketRole != EicFunction.BalanceResponsibleParty && marketRole != EicFunction.EnergySupplier)
+            {
+                meteringPointTypes.Add(new Option<MeteringPointType>(MeteringPointType.Exchange));
+                meteringPointTypes.Add(new Option<MeteringPointType>(MeteringPointType.TotalConsumption));
+            }
+
+            return meteringPointTypes;
+        }
+
+        public bool GetIsGridAreaRequired() => marketRole == EicFunction.GridAccessProvider;
```

Comment:
Question: What does this do, seems not to be called anywhere?

Reply:
I think so :-)

Reply:
b17d3a874f81ebb11efca2460ec92a0e8077ec9e